sekoia_automation/scripts/files_generator.py

```diff
@@ -174,8 +175,56 @@ def update_module_manifest(self, module: type[Module]):
             manifest = json.load(f)
 
         manifest["configuration"] = configuration_model.schema()
+        self.add_secrets_to_configuration(
+            manifest["configuration"], configuration_model.__fields__.values()
+        )
 
         with self.manifest_path.open("w") as out:
             out.write(json.dumps(manifest, indent=2))
 
         print(f"[green][+][/green] Updated {self.manifest_path}")
+
+    @staticmethod
+    def add_secrets_to_configuration(config, config_fields):
+        """Add module conf's secrets to the manifest, if any
+
+        If the module has secrets, they are handled by Pandytic as if
+        the parameter's name is "secret". We parse the model's fieds
+        to find if such secrets exist and if so, we remove the
+        "secret" field from the generated JSON and add fields with the
+        right names instead
+
+        Parameters
+        ----------
+        config : dict[str, Any]
+            Config generated by Pydantic from the Model
+        config_fields :
+            Raw list of fields in the module's Model
+        """
+        for field in config_fields:
+            if field.field_info.extra.get("secret", False):
+                name = field.name
+                config["properties"][name] = {
+                    "title": FilesGenerator.to_title(name),
+                    "type": "string",
+                }
```

Comment:
I think this is not required, `manifest["configuration"] = configuration_model.schema()` already took care of it
```suggestion
```

Reply:
done